### ISSUE TITLE
Moving forum post blank page

### ISSUE THREAD

**Issue by BigBoiJefe** [e107inc/e107#3619]
When I try and move a forum thread the page that I get redirected to is blank.  Any idea what's causing this?  I am running the most current github branch.  Thanks!

**Comment by Moc** [comment]
PHP version?

**Comment by Moc** [comment]
Nevermind, found what is causing the issue. Will get back to this.

**Comment by Moc** [comment]
Just making some notes while trying to figure this out. Related to userclass permissions issue:
https://github.com/e107inc/e107/issues/3525 and https://github.com/e107inc/e107/issues/3550

Clean install:
- Main Site Administrator (created during install) has no value for `user_class` in user table
- https://github.com/e107inc/e107/blob/4becf60d2896e22c6721d1829b8a9f8799d28ddc/e107_plugins/forum/forum_post.php#L70 returns empty 
- https://github.com/e107inc/e107/blob/4becf60d2896e22c6721d1829b8a9f8799d28ddc/e107_plugins/forum/forum_class.php#L417 query gets no results.
- `forum_moderator` set to 254
- e_UC_MAINADMIN is 250 
- e_UC_ADMIN is 254

**Comment by Moc** [comment]
@phibel You did a lot of work on this specific area. Could you look into why a Main Site Administrator is not able to move forum posts on a clean install?

**Comment by Moc** [comment]
I think this needs changing: 

https://github.com/e107inc/e107/blob/4becf60d2896e22c6721d1829b8a9f8799d28ddc/e107_plugins/forum/forum_post.php#L70

to:

`$moderatorUserIds = $forum->getModeratorUserIdsByPostId($this->id);`

**Comment by SimSync** [comment]
`getModeratorUserIdsByPostId($this->id)` doesn't work, because `$this->id` contains the threadid and not the postid.
Switched to `getModeratorUserIdsByThreadId($this->id)` to receive the moderators for the thread. This worked and included also the userid of the main admin.

**Comment by phibel** [comment]
All my test users were a member of the moderator class of the thread/forum, so I think thats why did not run into this problem. I will try to look at it until Wednesday @Moc.

**Comment by BigBoiJefe** [comment]
Are you saying that i have to create a group called MODERATOR?

**Comment by BigBoiJefe** [comment]
Welp, that didn't work...

**Comment by BigBoiJefe** [comment]
So from seeing this, I can't have different groups managing specific forums.  That kinda defeats the point of having forum permissions, no?

**Comment by Moc** [comment]
@BigBoiJefe no, just hang on a little longer. There is a bug in the core code which causes this issue.

**Comment by SimSync** [comment]
I think my last commit should fix this issue. PR #3620 is submitted.

**Comment by phibel** [comment]
The solution from @SimSync looks good to me, I tested it in my installation.

**Comment by Moc** [comment]
Once the LANs are fixed, I'll merge in the fix.

**Comment by BigBoiJefe** [comment]
@moc confirmed working.  Thanks!
